genvalidity-bytestring/src/Data/GenValidity/ByteString.hs

```diff
@@ -15,10 +15,17 @@ import qualified Data.ByteString.Internal as SB
 import qualified Data.ByteString.Lazy as LB
 import qualified Data.ByteString.Lazy.Internal as LB
 
--- | WARNING: Unchecked ByteStrings are *seriously* broken.
+-- | WARNING: Unchecked ByteStrings are __seriously__ broken.
+--
 -- The pointer may still point to something which is fine, but
 -- the offset and length will most likely be complete nonsense.
 -- This will most-likely lead to segfaults.
+--
+-- This means that 'genUnchecked' will generate seriously broken 'ByteString' values.
```

Comment:
No matter what it is, if something is broken, it either needs to be fixed, or thrown away. See my comment below on why I think current implementation of this instance is useless.